src/lib/telemetry.rs

```diff
@@ -136,12 +136,13 @@ pub enum LogKey {
 
 impl LogKey {
     pub fn add_suffix(&self, suffix: &str) -> LogKey {
-        let s = format!("{}-{}", &self.to_string(), suffix);
-        let s_str = s.as_str();
+        let mut s = self.to_string();
```

Comment:
I believe the original line 139 allocates two strings: one to generate the string from the Enum key and a second to create the final formatted string. The new option modifies the same String using string slices (`&str`), which is more efficient.